### PR TITLE
Ports Exosuit Fabricator's TGUI

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

ports
tgstation/tgstation#52073

## Why It's Good For The Game

why not

## Changelog
:cl: Timberpoes 
add: Roboticists rejoice. NanoTrasen has heard your cries for help. They've seen CCTV footage (recorded by the AI) of you all staring enviously at geneticists, lording around their shiny, polished, usable DNA Consoles. Cry no more, for you can now please your Silicon Overlords and robust those Machine Cultist Chaplains in high definition glory with a beautiful new interface added to Exosuit Fabricators.
/:cl:

### CODE REVIEW

**Comment by jupyterkat** [comment]
image

**Comment by Effemu** [comment]
big thanks epicman

**Comment by Sinestia** [comment]
Oh dang, said I'd do it this weekend and some chad actually just went and did it beforehand

excellent job man

File: code/__DEFINES/exosuit_fab.dm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#define BORG_MODULE_ENGINEERING		(1<<4)
+/// Module is used for service Cyborgs specialization
+#define BORG_MODULE_SPECIALITY		(1<<5)
+
+/// Module is compatible with Ripley Exosuit models
+#define EXOSUIT_MODULE_RIPLEY		(1<<0)
+/// Module is compatible with Odyseeus Exosuit models
+#define EXOSUIT_MODULE_ODYSSEUS		(1<<1)
+/// Module is compatible with Clarke Exosuit models
+#define EXOSUIT_MODULE_CLARKE		(1<<2)
```
Comment (Sarchutar) [inline code/__DEFINES/exosuit_fab.dm]: We don't have the Clarke mech and the firefighter is missing

Comment (jupyterkat) [reply code/__DEFINES/exosuit_fab.dm]: Firefighter has the same module set as ripley

**Comment by jupyterkat** [comment]
Jeez we really need tgui 4.3 for the stacks

**Comment by francinum** [comment]
Holy crap this is ugly

**Comment by spockye** [comment]
the conflict grind

**Comment by bloons3** [comment]
suffer

File: code/game/mecha/equipment/tools/mining_tools.dm [hunk 2]
```diff
@@ -150,16 +151,23 @@
...
 
 /obj/item/mecha_parts/mecha_equipment/mining_scanner/Initialize()
 	. = ..()
 	START_PROCESSING(SSfastprocess, src)
 
+/obj/item/mecha_parts/mecha_equipment/mining_scanner/can_attach(obj/mecha/M as obj)
+	if(..())
+		if(istype(M, /obj/mecha/working))
+			return TRUE
+	return FALSE
```
Comment (utilbird) [inline code/game/mecha/equipment/tools/mining_tools.dm]:
```suggestion
	return ..() && istype(M, /obj/mecha/working)
```

**Comment by jupyterkat** [comment]
PLEASE

**Comment by bloons3** [comment]
> PLEASE

can't you just unstage your tgui bundle, since it is built on commit/merge anyways?

**Comment by jupyterkat** [comment]
> > PLEASE
> 
> can't you just unstage your tgui bundle, since it is built on commit/merge anyways?

Then it'll fail the lint and won't get merged because lint check pass is required on protected branches

**Review by utilbird — APPROVED** [review]

**Comment by badlypayedgoon** [comment]
As dumb as that question might sound but: is there any chance this merge might be responsible for the "Upgrade mounting error! No suitable hardpoint detected!" borg module upgrade bug?

**Comment by bloons3** [comment]
> As dumb as that question might sound but: is there any chance this merge might be responsible for the "Upgrade mounting error! No suitable hardpoint detected!" borg module upgrade bug?

Make an issue on the github

**Pull request merged**